java-squid/src/main/java/org/sonar/java/se/symbolicvalues/SymbolicValueRelation.java

```diff
@@ -64,24 +83,40 @@ private boolean equalsRelation(SymbolicValueRelation relation) {
   }
 
   private int[] key() {
-    if (v1.id() < v2.id()) {
-      return new int[] {v1.id(), v2.id()};
+    if (leftOp.id() < rightOp.id()) {
+      return new int[] {leftOp.id(), rightOp.id()};
     } else {
-      return new int[] {v2.id(), v1.id()};
+      return new int[] {rightOp.id(), leftOp.id()};
     }
   }
 
   @Override
   public String toString() {
     StringBuilder buffer = new StringBuilder();
-    buffer.append(v1);
-    buffer.append(' ');
+    buffer.append(leftOp);
     buffer.append(getOperand());
-    buffer.append(' ');
-    buffer.append(v2);
+    buffer.append(rightOp);
     return buffer.toString();
   }
 
+  @Override
+  @CheckForNull
+  protected ProgramState checkRelation(BooleanConstraint booleanConstraint, ProgramState programState) {
+    Boolean fulfilled = impliedBy(programState.getKnownRelations());
+    if (fulfilled != null) {
+      if (BooleanConstraint.TRUE.equals(booleanConstraint)) {
+        if (!fulfilled) {
+          return null;
+        }
+      } else if (BooleanConstraint.FALSE.equals(booleanConstraint)) {
+        if (fulfilled) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Merge this if statement with the enclosing one. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1066)